docarray/data/torch_dataset.py

```diff
@@ -0,0 +1,73 @@
+from typing import Callable, Dict, Generic, List, Optional, Type, TypeVar
+
+from torch.utils.data import Dataset
+
+from docarray import BaseDocument, DocumentArray
+from docarray.typing import TorchTensor
+
+T_doc = TypeVar('T_doc', bound=BaseDocument)
+
+
+class MultiModalDataset(Dataset, Generic[T_doc]):
+    """
+    A dataset that can be used inside a PyTorch DataLoader.
+    In other words, it implements the PyTorch Dataset interface.
+
+    :param da: the DocumentArray to be used as the dataset
+    :param preprocessing: a dictionary of field names and preprocessing functions
+    """
+
+    document_type: Optional[Type[BaseDocument]] = None
+    __typed_ds__: Dict[Type[BaseDocument], Type['MultiModalDataset']] = {}
+
+    def __init__(
+        self, da: 'DocumentArray[T_doc]', preprocessing: Dict[str, Callable]
+    ) -> None:
+        self.da = da
+        self._preprocessing = preprocessing
```

Comment:
I think you already mentioned in the daily that you need to add documentation, let's make sure that we clearly convey the `k`s and `v`s of `preprocessing`, the inputs and outputs of the functions in its `v`s, etc.

Reply:
Yup. It became kind of a long docstring but I suppose that it ok